cli/flags.rs

```diff
@@ -951,6 +954,11 @@ update to a different location, use the --output flag
         .short("f")
         .help("Replace current exe even if not out-of-date"),
     )
+    .arg(
+      Arg::with_name("canary")
+        .long("canary")
+        .help("Upgrade to canary builds"),
```

Comment:
IMO this arg should take optional value that allows to specify concrete commit

Reply:
with the current state of the PR, that would be passed through `version`